### PR TITLE
Confirmation prompts

### LINKED ISSUE #84: As a user who modifies content, I want confirmation prompts, so that I can worry less about performing accidental operations
### 📝 Description

Currently, pressing the delete button anywhere will just go ahead with deleting instead of asking for confirmation first.

### ✅ Acceptance Criteria

- [ ] Add confirmation prompts to potentially destructive actions

### 💡 Additional context

_No response_

### PR DESCRIPTION
**Pull request by GAsplund**
Adds confirmation prompts to (hopefully) all delete buttons.
Closes #84

### CODE REVIEW

File: src/app/[locale]/settings/sponsors/RemoveSponsorButton.tsx [hunk 1]
```diff
@@ -10,11 +10,11 @@ const RemoveSponsorButton = ({
     id: number;
   };
 }) => {
-  return (
-    <ActionButton onClick={() => removeSponsor(sponsor.id)}>
-      Ta bort
-    </ActionButton>
-  );
+  const remove = async () => {
+    confirm('Are you sure you want to delete this sponsor?') &&
```
Comment (Oscariremma) [inline src/app/[locale]/settings/sponsors/RemoveSponsorButton.tsx:14]: Should this be hardcoded? Or come from i18n?

File: src/components/GammaGroupListItem/GammaGroupListItem.tsx [hunk 2]
```diff
@@ -17,11 +17,12 @@ const GammaGroupListItem = ({
   const router = useRouter();
 
   const remove = async () => {
-    await toast.promise(removeGroup(superGroupId), {
-      pending: 'Removing group...',
-      success: 'Group removed!',
-      error: 'Failed to remove group'
-    });
+    confirm('Are you sure you want to delete this group?') &&
```
Comment (Oscariremma) [inline src/components/GammaGroupListItem/GammaGroupListItem.tsx:20]: Again hadcoded or i18n?

File: src/app/[locale]/settings/notifiers/RemoveNotifierButton.tsx [hunk 3]
```diff
@@ -7,11 +7,12 @@ import { toast } from 'react-toastify';
 const RemoveNotifierButton = ({ id }: { id: number }) => {
   const remove = async () => {
     try {
-      await toast.promise(removeNotifier(id), {
-        pending: 'Removing notifier...',
-        success: 'Notifier removed!',
-        error: 'Failed to remove notifier'
-      });
+      confirm('Are you sure you want to delete this notifier?') &&
```
Comment (Oscariremma) [inline src/app/[locale]/settings/notifiers/RemoveNotifierButton.tsx:10]: Hardcoded or i18n? Technically none of the settings pages use i18n so maybe this is for a later fix?

**Review by Oscariremma — CHANGES_REQUESTED** [review]

**Comment by GAsplund** [comment]
The reason I did not include any i18n for those confirm prompts was because the toast did not either. They should both be implemented later when i18n for settings is implemented.

**Review by Oscariremma — APPROVED** [review]

**Pull request merged**